tests/test_env.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""Tests for context managers related to environment variables."""
+
+import pytest
+
+from macaron.env import get_patched_env
+
+
+@pytest.mark.parametrize(
+    ("before", "patch", "after"),
+    [
+        pytest.param(
+            {"PATH": "/usr/local/bin"},
+            {},
+            {"PATH": "/usr/local/bin"},
+            id="patch is empty",
+        ),
+        pytest.param(
+            {"PATH": "/usr/local/bin"},
+            {"GIT_TERMINAL_PROMPT": "0"},
+            {
+                "PATH": "/usr/local/bin",
+                "GIT_TERMINAL_PROMPT": "0",
+            },
+            id="patch adding a variable",
+        ),
+        pytest.param(
+            {"GIT_TERMINAL_PROMPT": "1"},
+            {"GIT_TERMINAL_PROMPT": "0"},
+            {"GIT_TERMINAL_PROMPT": "0"},
+            id="patch overriding a variable",
+        ),
+        pytest.param(
+            {"GIT_TERMINAL_PROMPT": "0"},
+            {"GIT_TERMINAL_PROMPT": None},
+            {},
+            id="patch removing a variable",
+        ),
+    ],
+)
+def test_patched_env(
+    before: dict[str, str],
+    patch: dict[str, str | None],
+    after: dict[str, str],
+) -> None:
+    """Tests for the ``patched_env`` context manager."""
```

Comment:
Need to update here too.

Reply:
Fixed in https://github.com/oracle/macaron/pull/389/commits/e5d50e65d823731d54b72edc5787b3171ddb6663